USER_MANUAL.md

```diff
@@ -1,12 +1,355 @@
+
+### Manual Objectives
+
+As mentioned in the _Why K?_ section above, the K Framework is designed as a
+collection of language-generic command-line interface (CLI) tools which revolve
+around K specifications. These tools cover a broad range of uses, but they
+typically fall into one of the following categories:
+
+1.  Transforming K Specs (e.g. compilation)
+2.  Running K Specs (e.g. concrete and symbolic execution)
+3.  Analyzing K Specs (e.g. theorem proving)
+
+The main *user-facing* K tools include:
+
+-   `kompile` - the K compiler driver
+-   `kast` - the stanadlone K parser and abstract syntax tree (AST)
+    transformation tool
+-   `krun` - the K interpreter and symbolic execution engine driver
+-   `kprove` - the K theorem prover
+
+This user manual is designed to be a tool reference.
+In particular, it is not desgined to be a tutorial on how to write K
+specifications or to teach the logical foundations of K. New K users should
+consult our dedicated
+[K tutorial](https://kframework.org/k-distribution/k-tutorial/),
+or the more language-design oriented
+[PL tutorial](https://kframework.org/k-distribution/pl-tutorial/).
+Researchers seeking to learn more about the logic underlying K are encouraged
+to peruse the
+[growing literature on K and matching logic](https://fsl.cs.illinois.edu/projects/pl/index.html).
+We will consider the manual complete when it provides a complete description of
+all user-facing K tools and features.
+
+Introduction to K
+-----------------
 
-This document contains documentation that has been written up to some extent
-but still needs to be ultimately included in the K manual which has not been
-written yet. New features of K that affect the surface language should be added
-to this document.
+Since K specifications are the primary input into the entire system, let us
+take a moment to describe them. At the highest level, K specifications describe
+a programming language or system using three different pieces:
+
+1.  the *system primitives*, the base datatypes used during system operation,
+    e.g., numbers, lists, maps, etc;
+2.  the *system state*, a tuple or record over system primitives which gives a
+    complete snapshot of the system at any given moment;
+3.  the *system behavior*, a set of rules which defines possible system
+    evolutions.
+
+K specifications are then defined by a collection of *sentences* which
+correspond to the three concepts above:
+
+1.  `syntax` declarations encode the *system primitives*;
+2.  `configuration` declarations encode the *system state*;
+3.  `context` and `rule` declarations encode the *system behavior*.
+
+K sentences are then organized into one or *modules* which are stored in one or
+more *files*. In this scheme, files may *require* other files and modules may
+*import* other modules, giving rise to a hierarchy of files and modules. We
+give an intuitive sketch of the two levels of grouping in the diagram below:
+
+```k
```

Comment:
this is not k code so the `k` label is not appropriate on this code fence

Reply:
Removed.